### PR TITLE
Use default holdability for prepareStatement + Tests

### PR DESCRIPTION
**Pull request by anandshah123**
Issue #63 
Enhances https://github.com/opendistro-for-elasticsearch/sql-jdbc/pull/64

*Description of changes:*
Fixed issue & added test cases to cover all `connection.prepareStatement` methods with `@ParameterizedTest`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/jdbc/ConnectionImpl.java [hunk 1]
```diff
@@ -279,7 +279,7 @@ private void validateResultSetHoldability(int holdability) throws SQLException {
     public PreparedStatement prepareStatement(String sql, int resultSetType, int resultSetConcurrency) throws SQLException {
         log.debug(() -> logEntry("prepareStatement (%s, %d, %d)", sql, resultSetType, resultSetConcurrency));
         checkOpen();
-        validateResultSetCharacteristics(resultSetType, resultSetConcurrency, resultSetConcurrency);
+        validateResultSetCharacteristics(resultSetType, resultSetConcurrency, ResultSet.HOLD_CURSORS_OVER_COMMIT);
```
Comment (penghuo) [inline src/main/java/com/amazon/opendistroforelasticsearch/jdbc/ConnectionImpl.java:282]: Could it be changed by using getHoldability()? Ref [JavaDoc](https://docs.oracle.com/javase/7/docs/api/java/sql/Connection.html#prepareStatement(java.lang.String,%20int,%20int))
> The holdability of the created result sets can be determined by calling getHoldability().

File: src/test/java/com/amazon/opendistroforelasticsearch/jdbc/PreparedStatementTests.java [hunk 2]
```diff
@@ -38,34 +36,98 @@
 @ExtendWith(PerTestWireMockServerExtension.class)
 public class PreparedStatementTests {
 
+    final String sql = "select pickup_datetime, trip_type, passenger_count, " +
+            "fare_amount, extra, vendor_id from nyc_taxis LIMIT 5";
+
+    private static Stream<Arguments> getArgumentsStream() {
+        int[] resultSetTypes = new int[]{ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.TYPE_FORWARD_ONLY};
+        int[] resultSetConcurrencyTypes = new int[]{ResultSet.CONCUR_READ_ONLY, ResultSet.CONCUR_UPDATABLE, ResultSet.CONCUR_READ_ONLY};
```
Comment (penghuo) [inline src/test/java/com/amazon/opendistroforelasticsearch/jdbc/PreparedStatementTests.java]: Duplicate ResultSet.CONCUR_READ_ONLY in the array.

File: src/test/java/com/amazon/opendistroforelasticsearch/jdbc/PreparedStatementTests.java [hunk 3]
```diff
@@ -1,32 +1,30 @@
...
-
 import com.amazon.opendistroforelasticsearch.jdbc.config.ConnectionConfig;
 import com.amazon.opendistroforelasticsearch.jdbc.logging.NoOpLogger;
-import com.amazon.opendistroforelasticsearch.jdbc.protocol.ConnectionResponse;
-import com.amazon.opendistroforelasticsearch.jdbc.protocol.JdbcQueryRequest;
-import com.amazon.opendistroforelasticsearch.jdbc.protocol.Protocol;
-import com.amazon.opendistroforelasticsearch.jdbc.protocol.ProtocolFactory;
-import com.amazon.opendistroforelasticsearch.jdbc.protocol.QueryRequest;
-import com.amazon.opendistroforelasticsearch.jdbc.protocol.QueryResponse;
+import com.amazon.opendistroforelasticsearch.jdbc.protocol.*;
```
Comment (zhongnansu) [inline src/test/java/com/amazon/opendistroforelasticsearch/jdbc/PreparedStatementTests.java]: Please avoid using wildcard import. You can enforce single import by configuring your IDE. Instructions for Intellij IDEA: https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#disable-wildcard-imports.

File: src/main/java/com/amazon/opendistroforelasticsearch/jdbc/ConnectionImpl.java (hunk 1 above)
Comment (anandshah123) [reply src/main/java/com/amazon/opendistroforelasticsearch/jdbc/ConnectionImpl.java:282]: No. JavaDoc says client can  determine holdability by calling `ResultSet.getHoldability()` which defaults to `ResultSet.HOLD_CURSORS_OVER_COMMIT ` in our case (Ref: https://github.com/opendistro-for-elasticsearch/sql-jdbc/blob/85112d09e0bbddfd831aa1e0801d335a0f957b1f/src/main/java/com/amazon/opendistroforelasticsearch/jdbc/ResultSetImpl.java#L1127)

**Review by penghuo — APPROVED** [review]
Thanks for the change.

**Review by dai-chen — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**